File: 2539-minimum-number-of-operations-to-make-arrays-similar/minimum-number-of-operations-to-make-arrays-similar.js
/**
 * @param {number[]} nums
 * @param {number[]} target
 * @return {number}
 */
var makeSimilar = function(nums, target) {
    // nums ve target'ı tek ve çift olarak ayır
    let numsOdd = nums.filter(x => x % 2 === 1).sort((a, b) => a - b);
    let numsEven = nums.filter(x => x % 2 === 0).sort((a, b) => a - b);
    let targetOdd = target.filter(x => x % 2 === 1).sort((a, b) => a - b);
    let targetEven = target.filter(x => x % 2 === 0).sort((a, b) => a - b);

    let ans = 0n;

    // Tekler için farkları hesapla
    for (let i = 0; i < numsOdd.length; i++) {
        ans += BigInt(Math.abs(numsOdd[i] - targetOdd[i]));
    }

    // Çiftler için farkları hesapla
    for (let i = 0; i < numsEven.length; i++) {
        ans += BigInt(Math.abs(numsEven[i] - targetEven[i]));
    }

    // Her işlem 2 birim değişiklik yaptığı için sonucu 4'e böl
    return Number(ans / 4n);
};

File: 2539-minimum-number-of-operations-to-make-arrays-similar/minimum-number-of-operations-to-make-arrays-similar.py
class Solution:
    def makeSimilar(self, nums, target):
        # Çift ve tekleri ayırıp sırala
        nums_even = sorted([x for x in nums if x % 2 == 0])
        nums_odd  = sorted([x for x in nums if x % 2 == 1])
        target_even = sorted([x for x in target if x % 2 == 0])
        target_odd  = sorted([x for x in target if x % 2 == 1])

        # Farkların toplamı
        total_diff = 0
        for a, b in zip(nums_even, target_even):
            total_diff += abs(a - b)
        for a, b in zip(nums_odd, target_odd):
            total_diff += abs(a - b)

        # Her işlem toplam farkı 4 azaltır
        return total_diff // 4

# Test (isteğe bağlı)
if __name__ == "__main__":
    print(Solution().makeSimilar([8,12,6], [2,14,10]))  # çıktı: 2
